nexus/src/external_api/params.rs

```diff
@@ -182,6 +182,20 @@ pub struct DiskCreate {
     pub size: ByteCount,
 }
 
+impl DiskCreate {
+    pub fn block_size(&self) -> u64 {
+        512
+    }
+
+    pub fn extent_size(&self) -> u64 {
+        1 << 20
```

Comment:
I wonder if (and now we are moving from code review to perhaps RFD territory), the block size should be decided at Nexus level, or if we pick the best defaults in crucible and Nexus does not have the ability to select block size.  

Do we yet know from a product point of view, do we have to support multiple block sizes for volumes?  Maybe we only need 4k and nothing else?  I don't know that either.

For extent_size, this extent size is not quite the same as what Crucible has.  This appears to be extent_size in bytes?  In Crucible, `extent_size` is the number of `block_size` parts that make up a single extent.  (It is not a good name, agreed).

Reply:
> I wonder if (and now we are moving from code review to perhaps RFD territory), the block size should be decided at Nexus level, or if we pick the best defaults in crucible and Nexus does not have the ability to select block size.
> 
> Do we yet know from a product point of view, do we have to support multiple block sizes for volumes? Maybe we only need 4k and nothing else? I don't know that either.

I dunno, but these are good questions. For now, I think the best thing Nexus can do - because it *has* to supply the values according to the existing API - is to "pick one thing and stick with it". That at least keeps the values in a well-known location, and is easy to change if/when we update policy.

> For extent_size, this extent size is not quite the same as what Crucible has. This appears to be extent_size in bytes? In Crucible, `extent_size` is the number of `block_size` parts that make up a single extent. (It is not a good name, agreed).

Oof, thanks for the heads up. I've filed https://github.com/oxidecomputer/crucible/issues/181 , I need to update the PR to fix this...

Reply:
> > I wonder if (and now we are moving from code review to perhaps RFD territory), the block size should be decided at Nexus level, or if we pick the best defaults in crucible and Nexus does not have the ability to select block size.
> > Do we yet know from a product point of view, do we have to support multiple block sizes for volumes? Maybe we only need 4k and nothing else? I don't know that either.
> 
> I dunno, but these are good questions. For now, I think the best thing Nexus can do - because it _has_ to supply the values according to the existing API - is to "pick one thing and stick with it". That at least keeps the values in a well-known location, and is easy to change if/when we update policy.

I think this is something where you want Crucible to advertise a supported range and have nexus select would makes sense so that way nexus is in charge of consistency across future revisions. In [RFD 29](https://rfd.shared.oxide.computer/rfd/0029) we speculate on block sizes. In think intent would be to start with 4k native devices, but I would not be surprised if we both need to deal with 512B devices over time to deal with older systems and want to suggest larger block sizes to guests to eventually deal with tings like QLC flash in a better way.

Reply:
Okay, this has been updated.

Reply:
@rmustacc : Filed https://github.com/oxidecomputer/crucible/issues/183 to track

Reply:
I have no further comments here, thanks for filing the issues.